dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -1125,18 +1125,24 @@ HRESULT WINAPI SHEmptyRecycleBinW(HWND hwnd, LPCWSTR pszRootPath, DWORD dwFlags)
         ret = RegGetValueW(HKEY_CURRENT_USER,
                            L"AppEvents\\Schemes\\Apps\\Explorer\\EmptyRecycleBin\\.Current",
                            NULL,
-                           RRF_RT_REG_SZ,
+                           RRF_RT_REG_SZ | RRF_RT_REG_EXPAND_SZ | RRF_NOEXPAND,
                            &dwType,
                            (PVOID)szPath,
                            &dwSize);
         if (ret != ERROR_SUCCESS)
             return S_OK;
 
-        if (dwType != REG_EXPAND_SZ) /* type dismatch */
+        if (dwType != REG_SZ && dwType != REG_EXPAND_SZ) /* Check whether the type is valid */
             return S_OK;
 
-        szPath[_countof(szPath)-1] = L'\0';
-        PlaySoundW(szPath, NULL, SND_FILENAME);
+        if (dwType == REG_EXPAND_SZ)
```

Comment:
In this whole code you could use `szBuffer` instead of `szPath`, and then do the expansion into `˙szPath`.
This would save space in the stack.

Reply:
OK, so as I understand, I can get rid of `szDest` at all.

Reply:
yep.

Reply:
Done, thanks.
Sound event still works correctly for me after this change.